test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1386 @@
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "mangle",
+        "-C",
+        "PREROUTING",
+        "-p",
+        "udp",
+        "-d",
+        "127.0.0.1",
+        "-m",
+        "state",
+        "-j",
+        "TEE",
+        "--gateway",
+        "192.168.10.1",
+        "-i",
+        "eth0",
+        "--destination-port",
+        "9521",
+        "--state",
+        "NEW",
+    ]
+
+
+@pytest.mark.parametrize(
+    "test_input",
+    [
+        'flags=ALL flags_set="ACK,RST,SYN,FIN"',
+        {"flags": "ALL", "flags_set": "ACK,RST,SYN,FIN"},
+        {"flags": ["ALL"], "flags_set": ["ACK", "RST", "SYN", "FIN"]},
+    ],
+)
+def test_tcp_flags(mocker, mock_basic_commands, test_input):
+    """Test various ways of inputting tcp_flags."""
+    rule_data = {
+        "chain": "OUTPUT",
+        "protocol": "tcp",
+        "jump": "DROP",
+        "tcp_flags": test_input,
+    }
+
+    set_module_args(rule_data)
+
+    commands_results = [
+        (0, "", ""),
+    ]
+
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-C",
+        "OUTPUT",
+        "-p",
+        "tcp",
+        "--tcp-flags",
+        "ALL",
+        "ACK,RST,SYN,FIN",
+        "-j",
+        "DROP",
+    ]
+
+
+@pytest.mark.parametrize(
+    "log_level",
+    [
+        "0", "1", "2", "3", "4", "5", "6", "7",
+        "emerg", "alert", "crit", "error", "warning", "notice", "info", "debug",
+    ],
+)
+def test_log_level(mocker, mock_basic_commands, log_level):
+    """Test various ways of log level flag."""
+
+    set_module_args(
+        {
+            "chain": "INPUT",
+            "jump": "LOG",
+            "log_level": log_level,
+            "source": "1.2.3.4/32",
+            "log_prefix": "** DROP-this_ip **",
+        }
+    )
+    commands_results = [
+        (0, "", ""),
+    ]
+
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-C",
+        "INPUT",
+        "-s",
+        "1.2.3.4/32",
+        "-j",
+        "LOG",
+        "--log-prefix",
+        "** DROP-this_ip **",
+        "--log-level",
+        log_level,
+    ]
+
+
+@pytest.mark.parametrize(
+    "test_input,expected",
```

Comment:
I find separate strings a bit nicer:
```suggestion
    ("test_input", "expected"),
```